src/workerd/jsg/jsvalue.h

```diff
@@ -504,6 +503,43 @@ inline JsObject Lock::opaque(T&& inner) {
   return JsObject(wrapped.template As<v8::Object>());
 }
 
+class JsFunction final: public JsBase<v8::Function, JsFunction> {
+ public:
+  using JsBase<v8::Function, JsFunction>::JsBase;
+
+  template <IsJsValue... Args>
+  JsValue call(Lock& js, const JsValue& recv, Args... args) const {
+    v8::Local<v8::Function> fn = *this;
+    v8::Local<v8::Value> argv[] = {args...};
+    return JsValue(check(fn->Call(js.v8Context(), recv, sizeof...(Args), argv)));
+  }
+
+  template <IsJsValue... Args>
+  JsValue callNoReceiver(Lock& js, Args... args) const {
+    return call(js, js.null(), kj::fwd<Args...>(args...));
+  }
+
+  template <IsJsValue... Args>
+  JsValue callGlobalReceiver(Lock& js, Args... args) const {
+    return call(js, js.global(), kj::fwd<Args...>(args...));
+  }
+
+  JsValue call(Lock& js, const JsValue& recv, v8::LocalVector<v8::Value>& args) const;
+  JsValue callNoReceiver(Lock& js, v8::LocalVector<v8::Value>& args) const;
+  JsValue callGlobalReceiver(Lock& js, v8::LocalVector<v8::Value>& args) const;
```

Comment:
Can you add a documentation to all the 3 call* methods explaining their use cases?